File: Array/Second Largest Element/second_largest_element.cpp
// Program to print the second largest element in array

#include<iostream>
#include<utility>
#include<algorithm>

using namespace std;

// function to scan the elements of array
void scan_array(int array[], int size)
{
    cout << "Enter the elements of the array: \n";

    for(int i=0; i<size; i++)
    {
        cout << "Element [" << i << "]: ";
        cin >> array[i];
    }
}

// function to print the array
void print_array(int array[], int size)
{
    cout << "Array: ";
    for(int i=0; i<size; i++)
    {
        cout << array[i] << " ";
    }
}

// function to compute and print the second largest element of array
void second_largest_element(int array[], int size)
{
    // sort the array
    std::sort(array, array+size);

    // printing the second largest element
    cout << "\nSecond Largest Element: " << array[size - 2];
}

// driver function
int main()
{
    int size, array[30];

    // scanning the number of elements of the array
    cout << "Enter the number of elemnts of the array: ";
    cin >> size;

    // calling function to scan the array
    scan_array(array, size);

    // calling function to print the array
    print_array(array, size);

    // calling the function to print the second largest element of the array
    second_largest_element(array, size);

    return 0;
}

File: Array/Second Largest Element/second_largest_element.py
# Program to print the second largest element in an array

# Scan the total number of elements
number_of_elements = int(input("Enter the number of elements of array: "))

# Scan the elements of array
print(f"Enter the {number_of_elements} elements of array separated by space: ", end="")
array = list(map(int, input().split()))[:number_of_elements]

# Sort the array
array.sort()

# Invert the sorted array
invert_array = array[::-1]

# Find the second largest element
for i in range(len(array)):
    if invert_array[i] < invert_array[0]:
        second_max_element = invert_array[i]
        break

# Print the second largest elements
print(f"Second largest Element: {second_max_element}")

